test/parallel/test-buffer-isutf8.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const { isUtf8 } = require('buffer');
+const { TextEncoder } = require('util');
+
+const encoder = new TextEncoder();
+
+assert.strictEqual(isUtf8(encoder.encode('hello')), true);
+assert.strictEqual(isUtf8(encoder.encode('ğ')), true);
+assert.strictEqual(isUtf8(Buffer.from([0xf8])), false);
```

Comment:
Can you add one or two CESU-8 inputs? https://en.wikipedia.org/wiki/CESU-8

And one or two overlong encodings? https://en.wikipedia.org/wiki/UTF-8 (grep for overlong)

The example from https://docs.rs/cesu8/latest/cesu8/ is both, by the way. :-)

Reply:
Thanks for the review. I'm updating my pull request.

Reply:
(It doesn't look like this comment has been addressed, so I'll unresolve it)

Reply:
I added couple of new tests. Can you validate, or propose new tests specifically to the requests? I might missed them again. (I'm not familiar with the edge examples)

Reply:
Yeah, the problem here is that the current tests are all checking the output of `TextEncoder.encode()`, which is per definition valid UTF-8 and will always return `true` when checked.

You'll want to check actual byte sequences as input here. For CESU-8 and overlong UTF-8, the Wikipedia pages linked above have examples of that; for WTF-8, you could use e.g. `ed a0 80`, which would correspond to `U+D800` if it were valid UTF-8.

All these cases should return `false` since they should be rejected by UTF-8 validators.

Reply:
Actually, all of these cases are also covered by test/fixtures/wpt/encoding/textdecoder-fatal.any.js. You could try to re-use that list.

Reply:
Thanks for the explanation. It helped a lot! I just updated it. Hopefully, this time, I cover these cases.